### PR TITLE
Move view note link and minimize the margin

### PR DESCRIPTION
**Pull request by amdomanska**
https://github.com/DOAJ/doajPM/issues/2904#issuecomment-998242478

image

@richard-jones simple js change.

@sssoz not sure whether that's what was meant by "close up the space", let me know if not. Also - I couldn't find a good place to add the css (text area has in general margin-bottom: 0.75 so to close up the space I just added -0.75 to the "view note" link). Could you move the inline style to some better place? Or maybe it can stay like this?

### CODE REVIEW

File: portality/static/js/formulaic.js [hunk 1]
```diff
@@ -1382,16 +1382,17 @@ var formulaic = {
             this.init = function() {
                 var viewClass = edges.css_classes(this.ns, "view");
                 var closeClass = edges.css_classes(this.ns, "close");
+                var viewDivClass = edges.css_classes(this.ns, "viewDiv")
```
Comment (richard-jones) [inline portality/static/js/formulaic.js:1385]: This class doesn't seem to get used

File: portality/static/js/formulaic.js [hunk 2]
```diff
@@ -1382,16 +1382,17 @@ var formulaic = {
...
+                this.divs = $("div[name='" + this.fieldDef["name"] + "__group']").find("textarea");
```
Comment (richard-jones) [inline portality/static/js/formulaic.js:1386]: Are we sure that replacing the `this.divs` member variable with a set of `textarea`s doesn't have any side effects in other methods that use the divs?

**Review by richard-jones — CHANGES_REQUESTED** [review]

File: portality/static/js/formulaic.js [hunk 3]
```diff
@@ -1382,16 +1382,17 @@ var formulaic = {
...
-                this.divs = $("div[name='" + this.fieldDef["name"] + "__group']");
                 for (var i = 0; i < this.divs.length; i++) {
                     var container = $(this.divs[i]);
                     var modalId = "modal-" + this.fieldDef["name"] + "-" + i;
 
                     var date = $("#" + this.fieldDef["name"] + "-" + i + "-note_date");
                     var note = $("#" + this.fieldDef["name"] + "-" + i + "-note");
 
-                    container.append(`<div><a href="#" class="` + viewClass + `">view note</a>
+                    $(`<button class="button button--cta ` + viewClass + `" style="margin: 0 1rem 1rem 0;">View note</button>
```
Comment (sssoz) [inline portality/static/js/formulaic.js:1395]: @amdomanska I’ve made some slight modifications to the styling here (explained in here: https://github.com/DOAJ/doajPM/issues/2904#issuecomment-1008761062)

**Review by sssoz — APPROVED** [review]

**Comment by sssoz** [comment]
@amdomanska I’ve merged this branch to mine, `feature/2904-notes-fix` (in https://github.com/DOAJ/doaj/pull/2017)

File: portality/static/js/formulaic.js (hunk 2 above)
Comment (amdomanska) [reply portality/static/js/formulaic.js:1386]: You're right, it shouldn't be done this way. I will simply use a new local var.

**Comment by amdomanska** [comment]
@richard-jones I have addressed your comments.
@sssoz After Richard will approve my latest changes feel free to merge my branch to yours again to have the most up to date version. Then we can close that pull request and leave just yours.

**Comment by sssoz** [comment]
Thanks @amdomanska ! 🤦‍♀️ I lost track of my own PRs and branches and didn’t realise I didn’t have a PR for this — it was just a blank, commit-less branch! Sorry for the confusion. We can simply keep things here in this PR

**Review by richard-jones — APPROVED** [review]

**Pull request merged**